pkg/services/auth/jwt_test.go

```diff
@@ -0,0 +1,33 @@
+package auth
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("JWT tokens", func() {
+
+	It("Verify should success", func() {
+
+		provider := "github"
+		token := "token"
+
+		jwtToken, err := Generate(SecretKey, time.Millisecond, provider, token)
+		Expect(err).NotTo(HaveOccurred())
+
+		claims, err := Verify(SecretKey, jwtToken)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(claims.Provider).To(Equal(provider))
+		Expect(claims.ProviderToken).To(Equal(token))
+
+		time.Sleep(time.Second)
+		claims, err = Verify(SecretKey, jwtToken)
+		Expect(err.Error()).To(MatchRegexp("invalid token: token is expired by.*"))
```

Comment:
Let's avoid checking a specific error message. Can we check for a specific error `type` instead?

Reply:
that sounds like a good idea, but not really sure how to do it when the error is dynamic 🤔 . That is why I'm using RegExp here. It could fail with `invalid token: token is expired by 4s` or `invalid token: token is expired by 3s`.

Reply:
There is no need to include the token expiration delta in the message. That might actually be a security risk.

Reply:
I'll replace the error msg coming from the jwt library with `unauthorized token`, not sure if that would be accurate 🤔

Reply:
Testing for another library's error string is a sure fire way to have flaky tests. If the other lib changes the error message by 1 character, unit tests will suddenly break once we upgrade that dependency.

Reply:
@jpellizzari I have receive feedback from James to assert error messages in previous PRs instead of just doing `.HaveOccurred()`. But I could be misunderstanding something. I'm not sure what I have to do here. @JamWils would you mind to help me clear this up ?

Reply:
Yes, I told @josecordaz to also verify error messages as well.  The example we were looking at when we chatted was you couldn't tell which tests were testing which error path.  By verifying the error messages you actually were reducing flakiness in the test.  I also prefer specificity with assertions.  It looks like we need to have a conversation as a team what the expectations are, pun intended.  Jordan I do agree with you about testing another library's error string.  To eliminate flakiness I would use a pattern such as:

```
func TestSomething(t *T) {
    ....
    err := MyDailyFunc("", "")
    
    expectedErr := json.Unmarshal(badData, &obj)
   t.Assert(err, Equals, fmt.Errorf("MyDailyFunc json unmarshal fail: %w, expectedErr))
}
```

In this instance, I actually propose we verify the string, not for my normal specificity reasons (error messages shouldn't change that often anyway), but for security reasons.  We want to lock-in the contract for the test and if someone changes the error message and puts sensitive information in it that we don't want returned then the test will fail.

We can revisit assertions outside of this PR.  I made my case but leave the final call up to you and Jose.

Reply:
I like the solution @josecordaz has now with keeping the assertion in sync with the error string. The important thing is to verify that the the error you are receiving is the one you expect, so we agree there.

I'm trying to to have tests break when trivial error messages changes happen. If as assertion has a string literal, you will have to update the assertion every time the error message changes. Having both the assertion AND the error message refer to the same string is :+1: